torchci/pages/test/[[...page]].tsx

```diff
@@ -0,0 +1,278 @@
+import dayjs from "dayjs";
+import {
+  Divider,
+  Grid,
+  Stack,
+  Typography,
+} from "@mui/material";
+import { useRouter } from "next/router";
+import { RocksetParam } from "lib/rockset";
+import { fetcher } from "lib/GeneralUtils";
+import useSWR from "swr";
+import { useState } from "react";
+import fetchS3Links from "lib/fetchS3Links";
+import { TimeSeriesPanelWithData } from "components/metrics/panels/TimeSeriesPanel";
+
+const LASTEST_N_RUNS = 50;
+const ROW_HEIGHT = 240;
+const TO_GB = 1024 * 1024 * 1024;
+
+function formatSeries(
+  data: any,
+) {
+  return {
+    type: "line",
+    symbol: "circle",
+    symbolSize: 4,
+    data,
+    emphasis: {
+      focus: "series",
+    },
+  };
+}
+
+function DisplayInsights({
+  data,
+  // Human-readable title of the panel.
+  title,
+  // What field name to treat as the time value.
+  timeFieldName,
+  // What field name to put on the y axis.
+  yAxisFieldName,
+  // What label to put on the y axis.
+  yAxisLabel,
+}: {
+  data: any;
+  title: string;
+  timeFieldName: string;
+  yAxisFieldName: string;
+  yAxisLabel?: string;
+}) {
+  const chartData = [];
+  for (const e of data) {
+    chartData.push([e[timeFieldName], e[yAxisFieldName]]);
+  }
+  const chartSeries = formatSeries(chartData);
+
+  return (
+    <Grid item xs={12} lg={6} height={ROW_HEIGHT}>
+      <TimeSeriesPanelWithData
+        data={chartData}
+        series={chartSeries}
+        title={title}
+        yAxisLabel={yAxisLabel}
+        yAxisRenderer={(unit) => `${unit}`}
+      />
+    </Grid>
+  );
+}
+
+function GetUsage({
+  jobName,
+  workflowName,
+  testFile,
+  testClass,
+  workflowIds,
+  jobIds,
+}: {
+  jobName: string;
+  workflowName?: string;
+  testFile?: string;
+  testClass?: string;
+  workflowIds: string[];
+  jobIds: string[];
+}) {
+  const params = {
+    workflowName: workflowName,
+    jobName: jobName,
+    testFile: testFile,
+    testClass: testClass,
+    workflowIds: workflowIds,
+    jobIds: jobIds,
+  };
+
+  const url = `/api/usage-log-aggregator/lambda?params=${encodeURIComponent(
+    JSON.stringify(params)
+  )}`;
+  const { data } = useSWR(url, fetcher);
+
+  if (data === undefined || data.length == 0) {
+    return (<div></div>);
+  }
+
+  const transformedData = [];
+  // Transform the data a bit to fit into the same rockset schema
+  for (var idx in data["timestamp"]) {
+    transformedData.push({
+      timestamp: data["timestamp"][idx],
+      cpu: data["cpu"][idx],
+      mem: data["mem"][idx] / TO_GB,
+      gpu: data["gpu"][idx],
+      gpu_mem: data["gpu_mem"][idx] / TO_GB,
+    });
+  }
+
+  return (
+    <Grid container spacing={2}>
```

Comment:
Can you add a small note here (that's visible on the hud page) explaining that the start and end of the time ranges on the charts roughly correspond to when the tests started and stopped running?

Reply:
I found a better way to express this by adding vertical lines with label `Start testing` and `Stop testing` into the chart